### PR TITLE
Mdicontainer iconchanges

### PR DESCRIPTION
**Pull request by gero3**
Fixes #4650

## Proposed changes

- cleanup code in accordance to @RussKie review in #4650

## Customer Impact

- None

## Regression? 

- No

## Risk

- None

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/4664)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs [hunk 1]
```diff
@@ -96,7 +94,7 @@ internal MenuStrip MergedMenu
             }
         }
 
-        private Image GetTargetWindowIcon()
+        private (Image Image,bool Visible) GetTargetWindowIcon()
         {
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:98]: @RussKie , was comment about not querying for Icon incase it was not set visible?

File: src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs [hunk 2]
```diff
@@ -105,7 +103,14 @@ private Image GetTargetWindowIcon()
...
+            return (systemIcon, _target is not Form formTarget || formTarget.ShowIcon);
+        }
+
+        public void UpdateIcon()
+        {
+            var imageInfo = GetTargetWindowIcon();
+            _system.Image = imageInfo.Image;
+            _system.Visible = imageInfo.Visible;
         }
         private bool GetTargetWindowIconVisibility() => _target is not Form formTarget || formTarget.ShowIcon;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:115]: nit: newline missing.

File: src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs [hunk 3]
```diff
@@ -150,9 +155,8 @@ private void OnTargetWindowHandleRecreated(object sender, EventArgs e)
                 _system.DropDown.Items.Clear();
                 _system.DropDown.Dispose();
             }
+            UpdateIcon();
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:158]: nit: enforcing empty newline after closing braces in the repo. If existing code doesn't follow, please feel free to make those cleanup changes as well.

File: src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs (hunk 1 above)
Comment (gero3) [reply src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:98]: I've made it so it doesn't load/ generate the image if the image isn't visible in the next commit.

**Comment by gero3** [comment]
@dreddy-work 

I've updated the PR with the remarks you gave me and I've also updated the `UpdateIcon` method to make sure that the previous image is disposed of.

File: src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs [hunk 4]
```diff
@@ -96,23 +94,33 @@ internal MenuStrip MergedMenu
             }
         }
 
-        private Image GetTargetWindowIcon()
+        private (Image Image,bool Visible) GetTargetWindowIcon()
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:97]: nit: missing format on this line.

File: src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs [hunk 5]
```diff
@@ -96,23 +94,33 @@ internal MenuStrip MergedMenu
...
         }
-        private bool GetTargetWindowIconVisibility() => _target is not Form formTarget || formTarget.ShowIcon;
 
-        public void updateIcon()
+        public void UpdateIcon()
         {
-            _system.Image = GetTargetWindowIcon();
-            _system.Visible = GetTargetWindowIconVisibility();
+            var imageInfo = GetTargetWindowIcon();
+            if (_system.Image is not null) {
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:119]: nit: braces always start on newline and a newline after closing braces

**Review by dreddy-work — COMMENTED** [review]
logically nothing should change but can you please do a integration test and confirm result with gif?

**Review by RussKie — CHANGES_REQUESTED** [review]

**Comment by RussKie** [comment]
Please rebase, resolve the conflicts, and address the feedback.

**Comment by dreddy-work** [comment]
@gero3, just want to make sure you are still on this.

**Comment by dreddy-work** [comment]
closing as there were no recent activity on this.

**Pull request closed without merging**